src/main/webapp/scripts/auth.js

```diff
@@ -24,36 +24,126 @@ const TUTORIAL_FILENAME = 'home';
 function getLogStatus(fileName) {
   fetch('/log-status').then(response => response.json()).then((logStatus) => {
     if (!logStatus.isRegistered && logStatus.isLogged) {
-      redirectToRegistrationForm();
+      showRegistrationForm(logStatus.isRegistered);
     }
     
     if (fileName === TUTORIAL_FILENAME) {
       handleTutorialContent(logStatus.isLogged);
     }
 
-    setLogURLInNavBar(logStatus);
+    handleLogForm(logStatus);
   });
 }
 
-function redirectToRegistrationForm() {
-  window.location.replace('./registration.html');
+/** @param {boolean} isRegistered */
+function showRegistrationForm(isRegistered) {
+  $('#registration-modal').load('common/registration.html', () => {
+    setUploadedImaged();
+    setIconClickEvent();
+    if (isRegistered) {
+      getUserInformation();
+    }
+    $('#exampleModalCenter').modal('show');
+  });
+}
+
+/** 
+* Retrieves the user information to set it in the registration form
+* @return {void}
+*/
+function getUserInformation() {
+  fetch('/get-vendor', {method: 'POST'}).then(
+    response => {
+      if (response.status === 401) {
+        response.text().then((error) => {
+          $('#log-button').trigger('click');
+          alert(error);
+          return;
+        });
+      } else if (response.status === 200) {
+        response.json()
+        .then((vendorInformation) => {
+          setVendorInformationInModal(vendorInformation);
+        });
+      }
+    }
+  );
+}
+
+// Sets the input event on the profile picture icon.
+function setIconClickEvent() {
+  $(".upload-button").on('click', function() {
+      $("#imageFile").click();
+  });
+}
+
+// Checks for uploaded files by the user to set the image or a placeholder
+function setUploadedImaged() {
+  $("#imageFile").change(function () {
+    if (this.files && this.files[0]) {
+      const reader = new FileReader();
+      reader.onload = function (e) {
+        $('#profile-picture').attr('src', e.target.result);
+      }
+      reader.readAsDataURL(this.files[0]);
+    } else {
+      $('#profile-picture').attr('src', 'images/placeholderImage.png');
+    }
+  });
+}
+
+// Sets the dropdown menu or URL link
+/** @param {{firstName:string, lastName:string, phoneNumber:string, blobKey:string, altText:string}} logStatus */
```

Comment:
Can we use the same JSDoc as in` line 38 `for all the functions ?
```suggestion
/** 
* @param {string} firstName
* @param {string} lastName 
* ...
*/
```

Reply:
For sure!